diff --git a/sound/soc/sunxi_v2/snd_sunxi_ahub.c b/sound/soc/sunxi_v2/snd_sunxi_ahub.c
index fcdecd695..7395409ea 100644
--- a/sound/soc/sunxi_v2/snd_sunxi_ahub.c
+++ b/sound/soc/sunxi_v2/snd_sunxi_ahub.c
@@ -113,7 +113,7 @@ static int sunxi_ahub_dai_set_pll(struct snd_soc_dai *dai,
 			return -EINVAL;
 		}
 	}
-	if (clk_set_rate(clk_info->clk_module, freq_out / 2)) {
+	if (clk_set_rate(clk_info->clk_module, freq_out)) {
 		SND_LOG_ERR(HLOG, "freq : %u module clk unsupport\n", freq_out);
 		return -EINVAL;
 	}
